src/sonic-yang-mgmt/sonic_yang.py

```diff
@@ -0,0 +1,450 @@
+import yang as ly
+import sys
+import json
+import glob
+
+"""
+Yang schema and data tree python APIs based on libyang python
+"""
+class sonic_yang:
+    def __init__(self, yang_dir):
+        self.yang_dir = yang_dir
+        self.ctx = None
+        self.module = None
+        self.root = None
+
+        try:
+            self.ctx = ly.Context(yang_dir)
+        except Exception as e:
+            print(e)
+
+    """
+    load_schema_module(): load a Yang model file
+    input:    yang_file - full path of a Yang model file
+    returns:  True - success   False - failed
+    """
+    def load_schema_module(self, yang_file):
+        try:
+            self.ctx.parse_module_path(yang_file, ly.LYS_IN_YANG)
+        except Exception as e:
+            print(e)
+            return False
+        return True
+
+    """
+    load_schema_module_list(): load all Yang model files in the list
+    input:    yang_files - a list of Yang model file full path
+    returns:  True - success   False - failed
+    """
+    def load_schema_module_list(self, yang_files):
+        for file in yang_files:
+            if (self.load_schema_module(file) == False):
+                return False
+        return True
+
+    """
+    load_schema_modules(): load all Yang model files in the directory
+    input:    yang_dir - the directory of the yang model files to be loaded
+    returns:  True - success   False - failed
+    """
+    def load_schema_modules(self, yang_dir):
```

Comment:
if yang_dir = self.yang_dir (always), then let's not pass it as arg.

Reply:
Not always,  you can load more Yang files from different directories